### PR TITLE
Add fix routine for L019

### LINKED ISSUE #453: Make comma placement (L019) a fixable lint
This _seems_ like it should be a pretty straightforward implementation -- place the commas before of after depending on the config?

Let me know if it's trickier than that -- if not, I'd be willing to take a stab at it. I think this would be a nice feature in a 0.4.0 release

Current behavior:

experiment.sql
```sql
select
    field,
    field1,
    field2
    , field3
from my_table
```

Lint ✔️ 
```sh
$ sqlfluff lint experiment.sql
== [experiment.sql] FAIL
L:   5 | P:   5 | L019 | Found leading comma. Expected only trailing.
```

Fix 🙅‍♂️ 
```sh
$ sqlfluff fix experiment.sql --rules L019
==== finding fixable violations ====
==== no fixable linting violations found ====
  [1 unfixable linting violations found]
```

### PR DESCRIPTION
**Pull request by pwildenhain**
Closes #453 

Probably easier to just look at the total diff, since the individual commits got a little messy as I was working through this 😬

### CODE REVIEW

File: src/sqlfluff/core/rules/std.py [hunk 1]
```diff
@@ -2066,33 +2067,116 @@ def _last_code_seg(raw_stack, idx=-1):
                 return raw_stack[idx]
             idx -= 1
 
-    def _eval(self, segment, raw_stack, **kwargs):
+    def _eval(self, segment, raw_stack, memory, **kwargs):
         """Enforce comma placement.
 
         If want leading commas, we're looking for trailing commas, so
```
Comment (NiallRees) [inline src/sqlfluff/core/rules/std.py:2073]: just a bit of wording to tidy up here (not your changes I realise)

**Comment by NiallRees** [comment]
Probably best @alanmcruickshank reviews this one as he knows the logic already.

**Comment by alanmcruickshank** [comment]
I'll review this this week. It's meaty enough that it may take me some time to pick through.

File: test/core/rules/rule_test_cases_test.py [hunk 2]
```diff
@@ -20,10 +20,9 @@
 
     y = yaml.safe_load(raw)
 
-    rule = y["rule"]
-    named_tests = {k: v for k, v in y.items() if k != "rule"}
-    ids.extend([rule + "_" + t for t in named_tests])
-    test_cases.extend([RuleTestCase(rule=rule, **v) for k, v in named_tests.items()])
+    rule = y.pop("rule")
+    ids.extend([rule + "_" + t for t in y])
+    test_cases.extend([RuleTestCase(rule=rule, **v) for k, v in y.items()])
```
Comment (NiallRees) [inline test/core/rules/rule_test_cases_test.py:25]: @pwildenhain hope you don't mind me jumping on your PR but wanted to tidy this up before I forgot!

**Review by alanmcruickshank — APPROVED** [review]
Took me a while to pick apart what you're doing with the `memory` in the rule, but it's a nice approach. I think there's a possibility that it will fail if we for whatever reason there's no code after the offending comma to allow the final `LintResult` lines to trigger from the memory, but I can only think of one example of where a query might end with a comma and it's pretty unlikely (Bigquery `SELECT` statement with trailing comma and no `FROM` clause).

**Comment by pwildenhain** [comment]
> Took me a while to pick apart what you're doing with the `memory` in the rule, but it's a nice approach.

Thanks, I really sat with this one for a while trying to figure out the best approach. Once I saw how you were utilizing `memory` in other fix routines, this is the solution that came to me 💡 

The bug you mentioned is def a possibility, though I think we can cross that bridge 🌉 if and when that issue comes up 🤞 

@NiallRees Appreciate you jumping in 🤜 🤛

**Pull request merged**